### PR TITLE
171088 - Stub for PKS API endpoint – To get permits for a given S-100 cell(s)

### PR DESCRIPTION
**Pull request by MastekRohitChettri**
This PR consists changes of [PBI 171088 ](https://dev.azure.com/ukhydro/S-100%20Permit%20Generator/_workitems/edit/171088)

- Code changes added

### CODE REVIEW

File: src/UKHO.S100PermitService.StubService/Stubs/ProductKeyServiceStub.cs [hunk 1]
```diff
@@ -8,21 +9,74 @@ namespace UKHO.S100PermitService.StubService.Stubs
 {
     public class ProductKeyServiceStub : IStub
     {
-        private readonly ProductKeyServiceConfiguration _productKeyServiceConfiguration;
```
Comment (jrippington) [inline src/UKHO.S100PermitService.StubService/Stubs/ProductKeyServiceStub.cs:11]: This is correct, should be '_productKeyServiceConfiguration'. Please amend the editor rule.

File: src/UKHO.S100PermitService.StubService/Stubs/ProductKeyServiceStub.cs [hunk 2]
```diff
@@ -8,21 +9,74 @@ namespace UKHO.S100PermitService.StubService.Stubs
...
+        public const string CONTENTTYPE = "Content-Type";
+        public const string APPLICATIONTYPE = "application/json";
+        private readonly ProductKeyServiceConfiguration productKeyServiceConfiguration;
 
         public ProductKeyServiceStub(ProductKeyServiceConfiguration productKeyServiceConfiguration)
         {
-            _productKeyServiceConfiguration = productKeyServiceConfiguration;
+            this.productKeyServiceConfiguration = productKeyServiceConfiguration;
```
Comment (jrippington) [inline src/UKHO.S100PermitService.StubService/Stubs/ProductKeyServiceStub.cs:18]: _productKeyServiceConfiguration = productKeyServiceConfiguration; <- is correct. Don't use 'this' here. Please amend the editor rule.

File: src/UKHO.S100PermitService.StubService/Stubs/ProductKeyServiceStub.cs [hunk 3]
```diff
@@ -8,21 +9,74 @@ namespace UKHO.S100PermitService.StubService.Stubs
...
+
+            server //404
+                  .Given(Request.Create()
+                  .WithPath(productKeyServiceConfiguration.Url)
+                  .UsingPost()
+                  .WithHeader("Authorization", "Bearer *", MatchBehaviour.AcceptOnMatch))
+                  .RespondWith(Response.Create()
+                        .WithStatusCode(HttpStatusCode.NotFound)
+                        .WithHeader(CONTENTTYPE, APPLICATIONTYPE)
+                        .WithBody("{\r\n  \"correlationId\": \"4c2dec2a-f44f-445d-837f-99fa72836533\",\r\n  \"errors\": [\r\n    {\r\n      \"source\": \"GetProductKey\",\r\n      \"description\": \"Key not found for Product and Edition \"\r\n    }\r\n  ]\r\n}\r\n"));
```
Comment (jrippington) [inline src/UKHO.S100PermitService.StubService/Stubs/ProductKeyServiceStub.cs:41]: Please define this JSON in a file as elsewhere

**Review by jrippington — CHANGES_REQUESTED** [review]
Few minor revisions, thanks!

File: src/UKHO.S100PermitService.StubService/Stubs/ProductKeyServiceStub.cs (hunk 1 above)
Comment (MastekRohitChettri) [reply src/UKHO.S100PermitService.StubService/Stubs/ProductKeyServiceStub.cs:11]: Resolved

File: src/UKHO.S100PermitService.StubService/Stubs/ProductKeyServiceStub.cs (hunk 2 above)
Comment (MastekRohitChettri) [reply src/UKHO.S100PermitService.StubService/Stubs/ProductKeyServiceStub.cs]: Resolved

File: src/UKHO.S100PermitService.StubService/Stubs/ProductKeyServiceStub.cs (hunk 3 above)
Comment (MastekRohitChettri) [reply src/UKHO.S100PermitService.StubService/Stubs/ProductKeyServiceStub.cs]: Moved

File: src/UKHO.S100PermitService.StubService/Stubs/ProductKeyServiceStub.cs [hunk 4]
```diff
@@ -8,6 +9,8 @@ namespace UKHO.S100PermitService.StubService.Stubs
 {
     public class ProductKeyServiceStub : IStub
     {
+        public const string CONTENTTYPE = "Content-Type";
+        public const string APPLICATIONTYPE = "application/json";
```
Comment (jrippington) [inline src/UKHO.S100PermitService.StubService/Stubs/ProductKeyServiceStub.cs:13]: As per dotnet_naming_style.constant_style.capitalization = pascal_case, please make these Pascal case?

**Review by jrippington — CHANGES_REQUESTED** [review]
Minor casing change, thanks

File: src/UKHO.S100PermitService.StubService/Stubs/ProductKeyServiceStub.cs (hunk 4 above)
Comment (MastekRohitChettri) [reply src/UKHO.S100PermitService.StubService/Stubs/ProductKeyServiceStub.cs]: Updated

**Review by jrippington — APPROVED** [review]
Looks good, thanks!

File: src/UKHO.S100PermitService.StubService/StubData/PKS/response-Allinvalid-404.json [hunk 5]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (harshal11869) [inline src/UKHO.S100PermitService.StubService/StubData/PKS/response-Allinvalid-404.json:1]: Can we please rename this file as response-datanotfound-404.json. Please set copy always property on file.

**Review by harshal11869 — CHANGES_REQUESTED** [review]

File: src/UKHO.S100PermitService.StubService/StubData/PKS/response-Allinvalid-404.json (hunk 5 above)
Comment (MastekRohitChettri) [reply src/UKHO.S100PermitService.StubService/StubData/PKS/response-Allinvalid-404.json]: Resolved

**Review by Nadeem14456 — APPROVED** [review]
LGTM.

**Review by jrippington — APPROVED** [review]
Looks good

**Pull request merged**